### ISSUE TITLE
kpt function writes kustomize resources to wrong path

### ISSUE THREAD

**Issue by satwell** [mgoltzsche/khelm#13]
I'm trying to use khelm to generate a kustomization directory for the traefik helm chart.  This mostly works, but some of the resources (`deployment_traefik.yaml` and `serviceaccount_traefik.yaml`) are getting written into the top level directory of the kpt package instead of the kustomization output dir.  Those files are referenced by `kustomization.yaml`, but the kustomization is broken because the files are in the wrong place.

To reproduce:
```shell
mkdir /tmp/traefik
kpt pkg init /tmp/traefik
cd /tmp/traefik
cat > function.yaml <<-EOF
apiVersion: v1
kind: ConfigMap
metadata:
  name: traefik-helm-output
  annotations:
    config.kubernetes.io/function: |
      container:
        image: mgoltzsche/khelm:2.0.0-beta.3
        network: true
    config.kubernetes.io/local-config: "true"
data:
  repository: https://helm.traefik.io/traefik
  chart: traefik
  version: v9.15.2
  name: traefik
  namespace: traefik
  outputPath: helmbase/
EOF
```

Then run:
```
/tmp/traefik% kpt fn run . --network
Running khelm 2.0.0-beta.3-9ac6a1b (helm 3.5.2)
WARNING: using untrusted repository "https://helm.traefik.io/traefik"
Downloading repository index of https://helm.traefik.io/traefik
Downloading chart traefik 9.15.2 from repo https://helm.traefik.io/traefik
Rendering chart traefik 9.15.2 with name "traefik" and namespace "traefik"
```

The generated helmbase/kustomization.yaml references `serviceaccount_traefik.yaml` and `deployment_traefik.yaml`:
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- customresourcedefinition_ingressroutes.traefik.containo.us.yaml
- customresourcedefinition_ingressroutetcps.traefik.containo.us.yaml
- customresourcedefinition_ingressrouteudps.traefik.containo.us.yaml
- customresourcedefinition_middlewares.traefik.containo.us.yaml
- customresourcedefinition_serverstransports.traefik.containo.us.yaml
- customresourcedefinition_tlsoptions.traefik.containo.us.yaml
- customresourcedefinition_tlsstores.traefik.containo.us.yaml
- customresourcedefinition_traefikservices.traefik.containo.us.yaml
- serviceaccount_traefik.yaml
- clusterrole_traefik.yaml
- clusterrolebinding_traefik.yaml
- deployment_traefik.yaml
- list.yaml
```

But these two files somehow ended up in the top level directory rather than in `helmbase/`.  So `kustomize build helmbase` fails:
```
Error: accumulating resources: accumulation err='accumulating resources from 'serviceaccount_traefik.yaml': evalsymlink failure on '/tmp/traefik/helmbase/serviceaccount_traefik.yaml' : lstat /tmp/traefik/helmbase/serviceaccount_traefik.yaml: no such file or directory': evalsymlink failure on '/tmp/traefik/helmbase/serviceaccount_traefik.yaml' : lstat /tmp/traefik/helmbase/serviceaccount_traefik.yaml: no such file or directory
```

I'm using khelm 2.0.0-beta.3 and kpt 0.38.0.

**Comment by satwell** [comment]
The bug is not limited to just kustomization output.  Setting `outputPath` to a single yaml file results in similar behavior.  The same two resources are written to `deployment_traefik.yaml` and `serviceaccount_traefik.yaml` instead of the output file.

**Comment by satwell** [comment]
I bisected the traefik helm repo and found that the issue first showed in up chart version v6.4.0 ([diff](https://github.com/traefik/traefik-helm-chart/compare/v6.3.0...v6.4.0)).  That change adds a null `annotations` to the ServiceAccount resource with the default chart values.  The Deployment also has a null `annotations`.  I suspect this is what's triggering the bug.

**Comment by mgoltzsche** [comment]
Hi @satwell, thanks for reporting the issue and pointing out the reason for it!
Indeed the empty `annotations` field is causing `kyaml.LookupCreate` not to create the field nor set annotations on it if it exists but is empty.
Your traefik example also shows another problem with khelm: `List` resources are not flattened which at least results in a weird output file and additional filters/transformations may not match the contained items.
I ll fix that.

**Comment by mgoltzsche** [comment]
This is fixed now in khelm 2.0.0-beta.4
